monkey/common/agent_events/hostname_discovery_event.py

```diff
@@ -0,0 +1,13 @@
+from . import AbstractAgentEvent
+
+
+class HostnameDiscoveryEvent(AbstractAgentEvent):
+    """
+    An event that occurs when the Agent identifies the hostname of the machine
+    on which it is running
```

Comment:
Doesn't matter if the agent discovered the hostname of the current machine or remote. `AbstractAgentEvent` has `target`. The hostname is of the target machine.

Reply:
`target` is `None` if it is the current machine.

Reply:
Maybe we should change the name of target to `remote_target` to clarify

Reply:
We can talk over webex. One way or another, its outside the scope of this PR.